src/alpaka/AlpakaCore/HistoContainer.h

```diff
@@ -0,0 +1,354 @@
+                                                                                    Histo *__restrict__ h) const {
+        const auto &[firstElementIdxGlobal, endElementIdxGlobal] =
+            cms::alpakatools::element_global_index_range_truncated(acc, Vec1::all(Histo::totbins()));
+
+        for (uint32_t i = firstElementIdxGlobal[0u]; i < endElementIdxGlobal[0u]; ++i) {
+          h->off[i] = 0;
+        }
+      }
+    };
+
+    template <typename Histo>
+    ALPAKA_FN_HOST ALPAKA_FN_INLINE __attribute__((always_inline)) void launchFinalize(
+        Histo *__restrict__ h,
+        const ALPAKA_ACCELERATOR_NAMESPACE::DevAcc1 &device,
+        ALPAKA_ACCELERATOR_NAMESPACE::Queue &queue) {
+      uint32_t *poff = (uint32_t *)((char *)(h) + offsetof(Histo, off));
+
+      const int num_items = Histo::totbins();
+
+      const unsigned int nthreads = 1024;
+      const Vec1 threadsPerBlockOrElementsPerThread(nthreads);
+      const unsigned int nblocks = (num_items + nthreads - 1) / nthreads;
+      const Vec1 blocksPerGrid(nblocks);
+
+      const WorkDiv1 &workDiv = cms::alpakatools::make_workdiv(blocksPerGrid, threadsPerBlockOrElementsPerThread);
+      alpaka::queue::enqueue(queue,
+                             alpaka::kernel::createTaskKernel<ALPAKA_ACCELERATOR_NAMESPACE::Acc1>(
+                                 workDiv, multiBlockPrefixScanFirstStep<uint32_t>(), poff, poff, num_items));
+
+      const WorkDiv1 &workDivWith1Block =
+          cms::alpakatools::make_workdiv(Vec1::all(1), threadsPerBlockOrElementsPerThread);
+      alpaka::queue::enqueue(
+          queue,
+          alpaka::kernel::createTaskKernel<ALPAKA_ACCELERATOR_NAMESPACE::Acc1>(
+              workDivWith1Block, multiBlockPrefixScanSecondStep<uint32_t>(), poff, poff, num_items, nblocks));
+    }
+
+    template <typename Histo, typename T>
+    ALPAKA_FN_HOST ALPAKA_FN_INLINE __attribute__((always_inline)) void fillManyFromVector(
+        Histo *__restrict__ h,
+        uint32_t nh,
+        T const *__restrict__ v,
+        uint32_t const *__restrict__ offsets,
+        uint32_t totSize,
+        unsigned int nthreads,
+        const ALPAKA_ACCELERATOR_NAMESPACE::DevAcc1 &device,
+        ALPAKA_ACCELERATOR_NAMESPACE::Queue &queue) {
+      const unsigned int nblocks = (totSize + nthreads - 1) / nthreads;
+      const Vec1 blocksPerGrid(nblocks);
+      const Vec1 threadsPerBlockOrElementsPerThread(nthreads);
+      const WorkDiv1 &workDiv = cms::alpakatools::make_workdiv(blocksPerGrid, threadsPerBlockOrElementsPerThread);
+
+      alpaka::queue::enqueue(
+          queue, alpaka::kernel::createTaskKernel<ALPAKA_ACCELERATOR_NAMESPACE::Acc1>(workDiv, launchZero(), h));
+
+      alpaka::queue::enqueue(queue,
+                             alpaka::kernel::createTaskKernel<ALPAKA_ACCELERATOR_NAMESPACE::Acc1>(
+                                 workDiv, countFromVector(), h, nh, v, offsets));
+      launchFinalize(h, device, queue);
```

Comment:
Following removal of `device`.
```suggestion
      launchFinalize(h, queue);
```

Reply:
Ha yes true, this argument is not necessary anymore now, thanks. I just removed it.